### ISSUE TITLE
ARM64: Assertion failed 'fieldSeq != FieldSeqStore::NotAField()'

### ISSUE THREAD

**Issue by kyulee1** [dotnet/coreclr#5954]
Assertion in a corefx test, System.Security.Cryptography.Pkcs.Tests
```
Assert failure(PID 9484 [0x0000250c], Thread: 18696 [0x4908]): Assertion failed 'fieldSeq != FieldSeqStore::NotAField()' in 'EncodeHelpers:EncodeRecipientId(ref,ref,long,long,ref):struct' (IL size 169)

    File: e:\github\coreclr\src\jit\valuenum.cpp Line: 2354
    Image: E:\Github\corefx\bin\tests\Windows_NT.AnyCPU.Release\System.Security.Cryptography.Pkcs.Tests\netcoreapp1.0\CoreRun.exe
```

**Comment by kyulee1** [comment]
I have a try fix -- https://github.com/dotnet/coreclr/compare/master...kyulee1:valseq?expand=1
which avoids such assertion earlier.
Anyhow will leave this to @briansull as it is.

**Comment by kyulee1** [comment]
This doesn't repro any more.

**Comment by kyulee1** [comment]
174 test results in corefx are not reported since this crashes xunit.

**Comment by briansull** [comment]
Testing a fix for this

**Comment by briansull** [comment]
Fixed with PR  #6589
